### PR TITLE
Fix clip frame drawing

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #1689

Also fixes clipping usage

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/TrackArtist.cpp [hunk 1]
```diff
@@ -301,20 +301,42 @@ void TrackArtist::UpdatePrefs()
 
 void TrackArt::DrawClipAffordance(wxDC& dc, const wxRect& rect, const wxString& title, bool highlight, bool selected)
 {
+   wxRect clipRect;
+   //Fix #1689: visual glitches appear on attempt to draw a rectangle
+   //larger than 0x7FFFFFF pixels wide (value was discovered
+   //by manual testing, and maybe depends on OS beeing used), but
```
Comment (Paul-Licameli) [inline src/TrackArtist.cpp:307]: "being"

File: src/TrackArtist.cpp [hunk 2]
```diff
@@ -301,20 +301,42 @@ void TrackArtist::UpdatePrefs()
...
+   //it's very unlikely that such huge rectangle will be ever fully visible
+   //on the screen, so we can safely reduce it's size to be slightly larger than
```
Comment (Paul-Licameli) [inline src/TrackArtist.cpp:309]: "its"

**Review by Paul-Licameli — APPROVED** [review]
Please fix comment typos.

File: src/TrackArtist.cpp (hunk 1 above)
Comment (vsverchinsky) [reply src/TrackArtist.cpp:307]: Thanks!

File: src/TrackArtist.cpp (hunk 2 above)
Comment (vsverchinsky) [reply src/TrackArtist.cpp:309]: Thanks!

**Pull request merged**